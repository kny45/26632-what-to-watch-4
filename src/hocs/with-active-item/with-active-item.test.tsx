import React from "react";
import renderer from "react-test-renderer";
import PropTypes from "prop-types";

import withActiveItem from "./with-active-item";

describe(`Render withActiveItem`, () => {
  const MockComponent = (props) => {
    const {children} = props;

    return (
      <div>
        {children}
      </div>
    );
  };

  MockComponent.propTypes = {
    children: PropTypes.oneOfType([
      PropTypes.arrayOf(PropTypes.node),
      PropTypes.node,
    ])
  };

  const MockComponentWrapped = withActiveItem(MockComponent);

  it(`withActiveItem is rendered correctly`, () => {
    const tree = renderer.create((
      <MockComponentWrapped
        activeItem = {false}
        onActiveItemChange = {() => {}}
      />), {
      createNodeMock() {
        return {};
      }
    }).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
