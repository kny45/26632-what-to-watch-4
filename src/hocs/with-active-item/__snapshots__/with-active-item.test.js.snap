// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render withActiveItem withActiveItem is rendered correctly 1`] = `<div />`;
